compiler/AST/expr.cpp

```diff
@@ -5479,7 +5487,7 @@ GenRet CallExpr::codegenPrimitive() {
       Type* dst = get(1)->typeInfo();
       Type* src = get(2)->typeInfo();
 
-      if (dst == src) {
+      if (dst == src && !(is_int_type(dst) || is_uint_type(dst)) ) {
```

Comment:
What I needed to have here is the size of C `int` for target architecture. C promotes smaller integers when some arithmetic is run on them. So following two snippets have different behaviors.

  ```
  uint8_t i1 = 255;
  uint8_t i2 = 1;
  uint8_t sum = i1+i2; // promote to int, sum, truncate... sum = 0

  if(sum) {
  }
  else {
    // we'll end up here
  }
  ```

  vs

  ```
  uint8_t i1 = 255;
  uint8_t i2 = 1;

  if(i1+i2) { // promote to int, sum... sum = 256
    // we'll end up here
  }
  else {
  }
  ```

If I could query size of `int` then this if should change to only include smaller-than-int integers. For now, I got stuck with all integers, as I couldn't figure out a simple/elegant way of finding size of `int`.